akka-http-core/src/main/scala/akka/http/engine/rendering/RenderSupport.scala

```diff
@@ -30,25 +29,24 @@ private object RenderSupport {
 
   val defaultLastChunkBytes: ByteString = renderChunk(HttpEntity.LastChunk)
 
-  // This hooks into the materialization to cancel the not needed second source. This helper class
-  // allows us to not take a FlowMaterializer but delegate the cancellation to the point when the whole stream
-  // materializes
-  private case class CancelSecond[T](first: Source[T], second: Source[T]) extends SimpleActorFlowSource[T] {
-    override def attach(flowSubscriber: Subscriber[T], materializer: ActorFlowMaterializer, flowName: String): Unit = {
-      first.to(Sink(flowSubscriber)).run()(materializer)
-      second.to(Sink.cancelled).run()(materializer)
-    }
+  def CancelSecond[T](first: Source[T, _], second: Source[T, _]): Source[T, Unit] = {
+    Source(first) { implicit b ⇒
+      frst ⇒
+        import FlowGraph.Implicits._
+        second ~> Sink.cancelled
+        frst.outlet
+    }.mapMaterialized((_) ⇒ ())
```

Comment:
this ensures unit as the result, but maybe it makes more sense here to propagate the first materialized value?